[공통] Redis 공통 모듈 생성

zmfpdl64 opened the pull request:
## 개요
공통으로 사용되는 Redis 설정
this branch is #200 

## PR 유형
어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [x] 코드 리팩토링
- [x] 테스트 추가, 테스트 리팩토링
- [x] 빌드 부분 혹은 패키지 매니저 수정
- [x] 파일 혹은 폴더명 수정
- [x] 파일 혹은 폴더 삭제

## PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.  Commit message convention 참고  (Ctrl + 클릭하세요.) 
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

> ### Description

- docker-compose.yml 작성
- Spring Redis Cluster 적용

### Redis docker 띄우기

## Redis 도커 컴포즈 실행
```bash
 docker-compose -p "redis-single"  -f ./redis-single.yml up -d
```
-p 도커 이름
-f 폴더 위치
-d 백그라운드 실행

## Redis 상태 확인
```bash
docker exec -it redis-single-redis_container-1 redis-cli -c -p 6379 info
```
## yml 파일 설정
```yml
spring:
  data:
    redis:
      host: localhost
      port: 6379
      redisson-prefix: "redis://"
```

zmfpdl64 committed:
Chore: redis 모듈 분리 및 docker-compose 작성

zmfpdl64 committed:
Feat: Redis Lettuce, Redisson 파일 작성

zmfpdl64 committed:
Refactor: Redis 모듈 분리 사이드 이펙트 변경

zmfpdl64 committed:
Refactor: bean 이름 추가

zmfpdl64 committed:
Refactor: 레디스 임베디드 클러스터로 변경

zmfpdl64 committed:
Refactor: Repository 테스트 삭제

zmfpdl64 committed:
Merge branch 'develop' of https://github.com/Team-Koreano/Koreano into feature/common/#200

dodari5882 commented on config/redis-master-1.conf:
```diff
@@ -0,0 +1,5 @@
+port 7001
```
레디스 클러스터 설정을 한파일로도 할수 있지 않나요??

zmfpdl64 replied:
다른 설정들은 같은데 port 번호는 달라서 설정파일을 각각 작성하게 됐습니다

dodari5882 commented on redis/src/main/java/org/ecommerce/redis/RedisApplication.java:
```diff
@@ -0,0 +1,14 @@
+import org.springframework.context.annotation.FullyQualifiedAnnotationBeanNameGenerator;
+
+@SpringBootApplication(nameGenerator = FullyQualifiedAnnotationBeanNameGenerator.class)
+public class RedisApplication {
```
redis는 application이 없는게 맞는듯 합니다.

zmfpdl64 replied:
모듈에서 실행파일은 없어야 될 것 같습니다 이부분 삭제하겠습니다

zmfpdl64 committed:
Refactor: Application클래스 제거

zmfpdl64 committed:
Refactor: 설정 추가

rdb파일이 정상적으로 들어가지 않았을 때 write 요청을 모두 차단하는 설정

allkey-lru는 모든 key를 lru알고리즘을 통해 삭제

zmfpdl64 committed:
Fix: idx name 수정

zmfpdl64 committed:
Refactor: 테스트 redis 포트번호 변경

zmfpdl64 committed:
Refactor: redis 단일서버로 테스트 변경

zmfpdl64 committed:
Revert "Refactor: redis 단일서버로 테스트 변경"

This reverts commit 5767a0dc0f1faf13b4af81b0d683b6d2f68d9c0d.

zmfpdl64 committed:
Fix: 임베디드 테스 Port 번호 변경

zmfpdl64 committed:
Refactor: redis Cluster 설정 제거

zmfpdl64 committed:
Refactor: Redis stand alone 적용

zmfpdl64 committed:
Chore: yml 설정

zmfpdl64 committed:
Merge branch 'develop' of https://github.com/Team-Koreano/Koreano into feature/common/#200

zmfpdl64 committed:
Fix: setting gradle 모듈 제거

zmfpdl64 committed:
Refactor:  redis 폴더 구조 변경
